### ISSUE TITLE
Test plan versions missing from data management

### ISSUE THREAD

**Issue by mcking65** [w3c/aria-at-app#760]
It looks like w3c/aria-at#980 should have created a new R&D version of V2023-08-23 for every test plan.  That version should be in the R&D column of data management. Yet, none of them show that version.

You can see [in the table of directories](https://github.com/w3c/aria-at/tree/master/tests) that every test plan was touched.

**Comment by mcking65** [comment]
@howard-e thank you for the additional detail about what is getting hashed. And, I apologize for overlooking the fact that v2023.08.23 was right there for several plans.

So, first, hats off to the Bocoup team because this is far smarter and better than I previously understood! It also gives me ideas on how we can build on these smarts with the V2 test format.

In particular, I've already developed a new approach to references so that we don't have to include URLs for spec references in the test files. Instead, I include the URL values in support.json. In the references, we only need the key for looking up the full URL. I've made this change in one of the branches I created where I'm working on our approach to refactoring test content. I still have to update the wiki and #977 with these details. So, as long as we don't resolve those URLs into the testPlanVersion.tests object, corrections to the URLs don't have to generate a new version of the plan.

This makes we further wonder about making yet another enhancement to the testPlanVersion object. What if we put all the data that we want hashed into the testPlanVersion.tests object and create another object, testPlanVersion.testsMetaData that includes the editorial content for each test. The data in testPlanVersion.testsMetaData would include the plan title, the title for each test, the presentationNumber for each test, the presentationNumber for each command, the setupScriptDescription for each setupScript, and the instructions for each test. This would mean that we may want to also track and display a last revision date for the metadata.

If I understand the system correctly, this would solve a ton of problems related to editorial content. Imagine that the action menubutton plan V2023.08.23 gets to recommended and someone discovers an editorial error. We could update any of that editorial content, and the new content would show up in the app and reports without moving a new version of the plan through the process. Is that correct? If so, wow, would that would be amazing!

**Comment by howard-e** [comment]
> So, first, hats off to the Bocoup team because this is far smarter and better than I previously understood! It also gives me ideas on how we can build on these smarts with the V2 test format.

Will be sure to pass it on!

> In particular, I've already developed a new approach to references so that we don't have to include URLs for spec references in the test files. Instead, I include the URL values in support.json. In the references, we only need the key for looking up the full URL. I've made this change in one of the branches I created where I'm working on our approach to refactoring test content. I still have to update the wiki and #977 with these details. So, as long as we don't resolve those URLs into the testPlanVersion.tests object, corrections to the URLs don't have to generate a new version of the plan.

Yep, this could definitely be supported.

> This makes we further wonder about making yet another enhancement to the testPlanVersion object. What if we put all the data that we want hashed into the testPlanVersion.tests object and create another object, testPlanVersion.testsMetaData that includes the editorial content for each test. The data in testPlanVersion.testsMetaData would include the plan title, the title for each test, the presentationNumber for each test, the presentationNumber for each command, the setupScriptDescription for each setupScript, and the instructions for each test. This would mean that we may want to also track and display a last revision date for the metadata.

We currently do have a `testPlanVersion.metadata` which could support what you're discussing here. Currently, it only holds  references to the pattern and example urls and other bits of the `references.csv`. Parts of what you'd like to include such as `setupScript`-related info has been a part of the individuals `testPlanVersion[].tests` object and intentionally ignored during the hashing. The idea of moving some of those attributes would certainly help understanding the shape of the data better and explicitly define the structure. And it also shouldn't affect the hashing procedure so this could be looked at! We could certainly explore that

> If I understand the system correctly, this would solve a ton of problems related to editorial content. Imagine that the action menubutton plan V2023.08.23 gets to recommended and someone discovers an editorial error. We could update any of that editorial content, and the new content would show up in the app and reports without moving a new version of the plan through the process. Is that correct? If so, wow, would that would be amazing!

Yep, if I also follow the scenario, it's certainly possible. It would mean just including the new data when the import script is ran and we should get that expected result.

**Comment by mcking65** [comment]
I think I mistakenly moved this back to in-progress. I've tested this in staging and now understand how it is working. This issue is resolved and can be closed after deployment.

I raised issue #801 to close the loop on the discussion of which fields to hash.

**Comment by ccanash** [comment]
Released
